### ISSUE TITLE
The library needs to be able to recover from errors

### ISSUE THREAD

**Issue by oleh-derevenko** [zaphoyd/websocketpp#896]
In case of communication or operation errors the library must provide means for client code to recover from the errors and return to normal operation. Writing to log only is insufficient. 
In particular, consider roles/server_endpoint.hpp line 187. 

        if (start_ec == error::async_accept_not_listening) {
            endpoint_type::m_elog->write(log::elevel::info,
                "Stopping acceptance of new connections because the underlying transport is no longer listening.");
        } else if (start_ec) { // <------------------------------------------------- HERE!
            endpoint_type::m_elog->write(log::elevel::rerror,
                "Restarting async_accept loop failed: "+ec.message());
        }

Here, after an error on accept continuation (for whatever reason) the code just writes a log and stops with no means to detect the error or recover from it.

**Comment by zaphoyd** [comment]
Yes, definitely a problem. As an asyncronous operation, `start_accept` really should take a callback to allow error delivery throughout the lifetime of the async loop.

**Comment by achataigner** [comment]
Hi,
I'm also very interested by this requirement.
Do you have any idea when it could be implemented ?
Best regards

**Comment by zaphoyd** [comment]
A comprehensive fix for this issue (as well as improvements to a related issue with the server version of get_connection having no way to get a detailed error code) has been pushed to the `develop` branch. start_accept now takes a handler that can return error information no matter when or why the async accept loop ends. A detailed error code from the transport layer is included as well to give client code the best chance to be able to do something useful with the error information.
